hudi-common/src/main/java/org/apache/hudi/common/model/HoodieFileGroup.java

```diff
@@ -77,7 +77,7 @@ public HoodieFileGroup(HoodieFileGroupId fileGroupId, HoodieTimeline timeline) {
     this.fileGroupId = fileGroupId;
     this.fileSlices = new TreeMap<>(HoodieFileGroup.getReverseCommitTimeComparator());
     this.timeline = timeline;
-    this.lastInstant = timeline.lastInstant();
+    this.lastInstant = timeline.filterCompletedAndCompactionInstants().lastInstant();
```

Comment:
We should avoid this. FileGroup should just be acting on the timeline given to make them composable.  Can you elaborate on why is there a need to ensure lastInstant must include only completed instants ?